pallets/gear/src/manager.rs

```diff
@@ -268,10 +268,19 @@ where
         Pallet::<T>::deposit_event(event);
     }
 
+    fn gas_recovered(&mut self, message_id: MessageId) {
+        let message_id = message_id.into_origin();
+        if let Ok(Some(gas_left)) = GasPallet::<T>::get_limit(message_id) {
+            log::debug!("Recovered: {:?} from: {:?}", gas_left, message_id);
+            GasPallet::<T>::decrease_gas_allowance(gas_left);
+            GasTree::<T>::remove(message_id);
```

Comment:
plz review this, not sure if there is any side effects for just removing the node

Reply:
Removing gas node from the tree must not be available. If there are unintended nodes left, it's an algorithmic error that we must correct. If I'm not mistaken, here could be improper side effects if gas less messages were sent in the program, before `exit` was called.

Reply:
> Generally, I think there is no need in a new journal note

`GasBurn` also have the ability to clean gas but it will run `repatriate_reserved` also which is not expected, what about adding a `message_id` in `ExitDispatch`, and clean the gas value inside `handler.exit_dispatch()` ?